ganttproject/src/biz/ganttproject/impex/csv/GanttCSVExport.java

```diff
@@ -85,44 +100,99 @@ public GanttCSVExport(IGanttProject project, CSVOptions csvOptions) {
     myHumanResourceManager = Preconditions.checkNotNull(resourceManager);
     myHumanResourceCustomPropertyManager = Preconditions.checkNotNull(resourceManager.getCustomPropertyManager());
     myRoleManager = Preconditions.checkNotNull(roleManager);
-    this.csvOptions = Preconditions.checkNotNull(csvOptions);
+    myCsvOptions = Preconditions.checkNotNull(csvOptions);;
+  }
+
+
+  private CSVFormat getFormatForWriter() {
+    CSVFormat format = CSVFormat.DEFAULT.withEscape('\\');
+    if (myCsvOptions.sSeparatedChar.length() == 1) {
+      format = format.withDelimiter(myCsvOptions.sSeparatedChar.charAt(0));
+    }
+    if (myCsvOptions.sSeparatedTextChar.length() == 1) {
+      format = format.withQuote(myCsvOptions.sSeparatedTextChar.charAt(0));
+    }
+
+    return format;
+  }
+
+  private SpreadsheetWriter getWriter(OutputStream stream, FileExtensionEnum extension) throws IOException {
+    if (extension != null ) {
+      switch (extension) {
+        case CSV:
+          return getCsvWriter(stream);
+        case XLS:
+          return getXlsWriter(stream);
+        default:
```

Comment:
Oh right, you need to return value from this method. Yes, leave IllegalArgumentException in the default.